### ISSUE TITLE
dependency issue: sqlparse 0.4.1-1

### ISSUE THREAD

**Issue by radio-van** [dbcli/pgcli#1218]
## Description
After updating **sqlparse** to 0.4.1-1, **pgcli** fails to run because of dependency issue.
It's obvious, because in **pgcli**'s requirement **sqlparse** is written as `sqlparse<4.0,>=0.3.0`.

I've tried to drop this restriction but then a couple of tests are failed:
`tests/test_smart_completion_public_schema_only.py`:
```python
test_specials_included_for_initial_completion

initial_text = '    '

    @pytest.mark.parametrize("initial_text", ("", "    ", "\t \t"))
    def test_specials_included_for_initial_completion(initial_text):
>       suggestions = suggest_type(initial_text, initial_text)

tests/test_sqlcompletion.py:821:

../../pgcli/packages/sqlcompletion.py:142: in suggest_type
    stmt = SqlStatement(full_text, text_before_cursor)
../../pgcli/packages/sqlcompletion.py:59: in __init__
    full_text, text_before_cursor, self.local_tables = isolate_query_ctes(
../../pgcli/packages/parseutils/ctes.py:23: in isolate_query_ctes
    ctes, remainder = extract_ctes(full_text)
_ _ _ 

sql = '    '

    def extract_ctes(sql):
        """ Extract constant table expresseions from a query

            Returns tuple (ctes, remainder_sql)

            ctes is a list of TableExpression namedtuples
            remainder_sql is the text from the original query after the CTEs have
            been stripped.
        """

>       p = parse(sql)[0]
E       IndexError: tuple index out of range

../../pgcli/packages/parseutils/ctes.py:58: IndexError
```

Of course the problem is gone if I use a separate venv, but I prefer to use AUR's version of **pgcli** to make it system-wide.
And because of rolling-release it's unconvinient to keep **sqlparse** outdated.
I hope this trouble with deps could be fixed)

## Your environment
ArchLinux 5.8.7-arch1-1 x86_64
bash 5.0.018-1
pgcli 3.0.0-1 (from AUR)
python-sqlparse 0.4.1-1 (from cimmunity)

**Comment by ShaheedHaque** [comment]
I would also be grateful if pgcli were to to unpin sqlparse 4.x because this prevents [pgAdmin from unpinning it](https://www.postgresql.org/message-id/CAHAc2jdxT%2B07rLkoQgb4NF_y4Sg46BFC0a3%3DKr_Mz%2BT_VcHwbw%40mail.gmail.com).

**Comment by amrishparmar** [comment]
With the [new release of pip 20.3](https://pythoninsider.blogspot.com/2020/11/pip-20-3-release-new-resolver.html) the new dependency resolver is now enabled by default, so this could cause failing builds in cases where sqlparse is pinned to >= 0.4.1.

A workaround for now is to pin to sqlparse 3.x, or use the legacy resolver (`--use-deprecated=legacy-resolver`), but that will only work until pip 21 (which is due in Jan '21), so would be ideal to have an official fix 🙏

**Comment by kevinmarsh** [comment]
This has been resolved on `master` but we'll need a new release to PyPi before most people will be able to resolve their dependency issues

**Comment by j-bennet** [comment]
released in 3.1.0:
https://github.com/dbcli/pgcli/blob/master/changelog.rst#310
